src/Sema.zig

```diff
@@ -12468,10 +12477,34 @@ fn zirReify(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
             const ty = try Type.Tag.error_set_merged.create(sema.arena, names);
             return sema.addType(ty);
         },
+        .Struct => {
+            // TODO use reflection instead of magic numbers here
+            const struct_val = union_val.val.castTag(.@"struct").?.data;
+            // layout: containerlayout,
+            const layout_val = struct_val[0];
+            // fields: []const enumfield,
+            const fields_val = struct_val[1];
+            // decls: []const declaration,
+            const decls_val = struct_val[2];
+            // is_tuple: bool,
+            const is_tuple_val = struct_val[3];
+
+            // Decls
+            const decls_slice_val = decls_val.castTag(.slice).?.data;
+            const decls_decl = decls_slice_val.ptr.pointerDecl().?;
+            try sema.ensureDeclAnalyzed(decls_decl);
+            if (decls_decl.ty.arrayLen() > 0) {
+                return sema.fail(block, src, "reified structs must have no decls", .{});
+            }
+
+            return if (is_tuple_val.toBool())
+                sema.reifyTuple(block, src, fields_val)
+            else
+                sema.reifyStruct(block, inst, src, layout_val, fields_val);
```

Comment:
It's good hygiene to put the `try` in each of the prongs here like this

```suggestion
                try sema.reifyTuple(block, src, fields_val)
            else
                try sema.reifyStruct(block, inst, src, layout_val, fields_val);
```

because in an error return trace, it will point at the `try` in the stack frame, telling you which prong was taken. This is similarly why you see `return error.Foo` in each prong of switch cases rather than `return switch(...`